python/cudf/cudf/_lib/groupby.pyx

```diff
@@ -167,6 +167,46 @@ cdef class GroupBy:
             included_aggregations_i = []
             col_aggregations = []
             for agg in aggs:
+                str_agg = str(agg)
```

Comment:
Question: Is it not enough to raise the `TypeError` if the `if valid_aggregations == "ALL" or agg_obj.kind in valid_aggregations:` condition below fails?

Reply:
Ideally yes, but that is a very loose condition for many operations vs dtype matrix we have and libcudf seems to be silently returning empty columns for those. We need to dig deeper for other types aswell.

Reply:
OK. I suppose this PR is a good enough intermediate solution until we can centeralize this in `get_valid_aggregation` or similar.